### PR TITLE
Fix project warnings

### PR DESCRIPTION
**Pull request by azabbasi**
https://msazure.visualstudio.com/One/_workitems/edit/8792109
https://msazure.visualstudio.com/One/_workitems/edit/8792111
https://msazure.visualstudio.com/One/_workitems/edit/8792122

### CODE REVIEW

**Review by timtay-microsoft — APPROVED** [review]

File: iothub/service/src/PurgeMessageQueueResult.cs [hunk 1]
```diff
@@ -21,6 +21,6 @@ public sealed class PurgeMessageQueueResult
         /// The total number of messages that were purged from the device's queue.
         /// </summary>
         [JsonProperty(PropertyName = "totalMessagesPurged", Required = Required.Always)]
-        public int TotalMessagesPurged;
```
Comment (timtay-microsoft) [inline iothub/service/src/PurgeMessageQueueResult.cs:24]: Is this a breaking change?

Comment (azabbasi) [reply iothub/service/src/PurgeMessageQueueResult.cs:24]: I don't believe so, public fields are both gettable and settable

File: iothub/device/src/TransientFaultHandling/ExponentialBackoff.cs [hunk 2]
```diff
@@ -23,55 +23,55 @@
...
-    /// A retry strategy with backoff parameters for calculating the exponential delay between retries.
+    /// A retry strategy with back-off parameters for calculating the exponential delay between retries.
     /// </summary>
     internal class ExponentialBackoff : RetryStrategy
     {
-        private readonly int retryCount;
+        private readonly int _retryCount;
 
-        private readonly TimeSpan minBackoff;
+        private readonly TimeSpan _minBackoff;
```
Comment (azabbasi) [inline iothub/device/src/TransientFaultHandling/ExponentialBackoff.cs:32]: should we rename these to `_minBackOff` to camelcase the O ??

Comment (abhipsaMisra) [reply iothub/device/src/TransientFaultHandling/ExponentialBackoff.cs:32]: https://en.wikipedia.org/wiki/Exponential_backoff
all lowercase

File: iothub/device/src/TransientFaultHandling/ExponentialBackoff.cs [hunk 3]
```diff
@@ -23,55 +23,55 @@
 namespace Microsoft.Azure.Devices.Client.TransientFaultHandling
 {
     /// <summary>
-    /// A retry strategy with backoff parameters for calculating the exponential delay between retries.
+    /// A retry strategy with back-off parameters for calculating the exponential delay between retries.
```
Comment (abhipsaMisra) [inline iothub/device/src/TransientFaultHandling/ExponentialBackoff.cs:26]: hyphenated?

Comment (azabbasi) [reply iothub/device/src/TransientFaultHandling/ExponentialBackoff.cs:26]: my spell checker doesn't like backoff as a single word ... if we all agree that it is a single word I will change these back.

File: iothub/service/src/ExportImportDevice.cs [hunk 4]
```diff
@@ -11,27 +11,51 @@
...
     public sealed class ExportImportDevice
     {
         private string _eTag;
         private string _twinETag;
 
         /// <summary>
-        /// making default ctor public
+        /// Property container
+        /// </summary>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1034:Nested types should not be visible", Justification = "Publicly property. No behavior changes allowed.")]
```
Comment (barustum) [inline iothub/service/src/ExportImportDevice.cs:24]: publicly -> public

File: iothub/service/src/ExportImportDevice.cs [hunk 5]
```diff
@@ -11,27 +11,51 @@
...
+        /// <summary>
+        /// Create an ExportImportDevice
         /// </summary>
         public ExportImportDevice()
         {
         }
 
         /// <summary>
-        /// ctor which takes a Device object along with import mode
+        /// Create an ExportImportDevice
```
Comment (barustum) [inline iothub/service/src/ExportImportDevice.cs:48]: use <see cref="ExportImportDevice" />

File: iothub/service/src/PurgeMessageQueueResult.cs [hunk 6]
```diff
@@ -2,10 +2,10 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // ---------------------------------------------------------------
 
+using Newtonsoft.Json;
```
Comment (barustum) [inline iothub/service/src/PurgeMessageQueueResult.cs:5]: why did you need to add this using?

**Review by abhipsaMisra — APPROVED** [review]

File: iothub/device/src/TransientFaultHandling/ExponentialBackoff.cs (hunk 3 above)
Comment (drwill-ms) [reply iothub/device/src/TransientFaultHandling/ExponentialBackoff.cs:26]: compound adjectives should be hyphenated

Comment (drwill-ms) [reply iothub/device/src/TransientFaultHandling/ExponentialBackoff.cs:26]: when used as a verb, compound phrases do not use hyphens

File: iothub/device/src/TransientFaultHandling/ExponentialBackoff.cs (hunk 2 above)
Comment (drwill-ms) [reply iothub/device/src/TransientFaultHandling/ExponentialBackoff.cs:32]: Wikipedia is not the authority on grammar and spelling tho

Comment (drwill-ms) [reply iothub/device/src/TransientFaultHandling/ExponentialBackoff.cs:32]: You could change the non-public variables, but I noticed there are parameters with those names and that would be a breaking change, so you might as well leave it the way it is so it is consistent.

File: iothub/service/src/ExportImportDevice.cs (hunk 4 above)
Comment (drwill-ms) [inline iothub/service/src/ExportImportDevice.cs:24]: Why did the class get moved?

File: iothub/service/src/ExportImportDevice.cs [hunk 7]
```diff
@@ -87,32 +111,35 @@ public string ETag
         [JsonProperty(PropertyName = "authentication")]
         public AuthenticationMechanism Authentication { get; set; }
 
+        /// <summary>
+        /// Twin ETag
```
Comment (drwill-ms) [inline iothub/service/src/ExportImportDevice.cs:115]: These need to be more meaningful. See our related comments in ADT for examples.

**Review by drwill-ms — APPROVED** [review]

File: iothub/service/src/ExportImportDevice.cs (hunk 4 above)
Comment (azabbasi) [reply iothub/service/src/ExportImportDevice.cs:24]: I just moved it to the top of the wrapper class.

File: iothub/service/src/PurgeMessageQueueResult.cs (hunk 6 above)
Comment (azabbasi) [reply iothub/service/src/PurgeMessageQueueResult.cs:5]: I didn't. Moved it from inside the namespace to outside

File: iothub/service/src/ExportImportDevice.cs (hunk 5 above)
Comment (azabbasi) [reply iothub/service/src/ExportImportDevice.cs:48]: That will be a pointer to the same file .. what's the benefit ?

Comment (barustum) [reply iothub/service/src/ExportImportDevice.cs]: True. It's just easier to make sure the comment is not outdated if we rename the class. I also see you used `<see cref="ExportImportDevice" />,` in the ExportImportDevice.cs class, so I thought maybe keep same pattern.

**Review by barustum — APPROVED** [review]

File: iothub/service/src/ExportImportDevice.cs [hunk 8]
```diff
@@ -11,27 +11,51 @@
...
+        /// <summary>
+        /// Create an ExportImportDevice <see cref="ExportImportDevice" />
         /// </summary>
         public ExportImportDevice()
         {
         }
 
         /// <summary>
-        /// ctor which takes a Device object along with import mode
+        /// Create an ExportImportDevice <see cref="ExportImportDevice" />
```
Comment (barustum) [inline iothub/service/src/ExportImportDevice.cs:48]: @azabbasi Wouldn't this show the 'ExportImportDevice' twice? I think it should be `Create an <see cref="ExportImportDevice" />`

Comment (drwill-ms) [reply iothub/service/src/ExportImportDevice.cs:48]: Yes, it will.
